lib/Controller/OCSController.php

```diff
@@ -148,30 +153,47 @@ public function createPublic(
 			$wopi = $this->tokenManager->newInitiatorToken($host, null, $shareToken, true, $this->userId);
 
 			$client = \OC::$server->getHTTPClientService()->newClient();
-			$response = $client->post(rtrim($host, '/') . '/ocs/v2.php/apps/richdocuments/api/v1/direct/share/initiator?format=json', [
-				'body' => [
-					'initiatorServer' => \OC::$server->getURLGenerator()->getAbsoluteURL(''),
-					'initiatorToken' => $wopi->getToken(),
-					'shareToken' => $shareToken,
-					'path' => $path,
-					'password' => $password
-				],
-				'timeout' => 30
-			]);
+			try {
+				$response = $client->post(rtrim($host, '/') . '/ocs/v2.php/apps/richdocuments/api/v1/direct/share/initiator?format=json', [
+					'body' => [
+						'initiatorServer' => \OC::$server->getURLGenerator()->getAbsoluteURL(''),
+						'initiatorToken' => $wopi->getToken(),
+						'shareToken' => $shareToken,
+						'path' => $path,
+						'password' => $password
+					],
+					'timeout' => 30
+				]);
+			} catch (\Exception $e) {
+				$response = new DataResponse([], HTTP::STATUS_FORBIDDEN);
+				$response->throttle();
+				return $response;
```

Comment:
Shouldn't the status be internal server error instead of forbidden?
Also, would logging the exception be useful? Same for the other exception handlers.